ironfish/src/wallet/account/encoder/bech32.ts

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { PUBLIC_ADDRESS_LENGTH } from '@ironfish/rust-nodejs'
+import bufio from 'bufio'
+import { Bech32m } from '../../../utils'
+import {
+  AccountImport,
+  KEY_LENGTH,
+  VERSION_LENGTH,
+  VIEW_KEY_LENGTH,
+} from '../../walletdb/accountValue'
+import { AccountEncoder } from './encoder'
+
+export const BECH32_ACCOUNT_PREFIX = 'ironfishaccount00000'
+
+export class Bech32AccountEncoder implements AccountEncoder {
+  encode(value: AccountImport): string {
+    const bw = bufio.write(this.getSize(value))
+    bw.writeU16(value.version)
+
+    let flags = 0
+    flags |= Number(!!value.spendingKey) << 0
+    flags |= Number(!!value.createdAt) << 1
+    bw.writeU8(flags)
+
+    bw.writeVarString(value.name, 'utf8')
+    if (value.spendingKey) {
+      bw.writeBytes(Buffer.from(value.spendingKey, 'hex'))
+    }
```

Comment:
can we move optional things to the end so the data / serialization looks stable in the case of a view wallet version of this? Then one is a superset of the other.